### PR TITLE
attestation: add base types for the attestation object and tests

### PR DESCRIPTION
**Pull request by calebdoxsey**
Add base types for the `AttestationObject`.

* Test data was taken from https://github.com/passwordless-lib/fido2-net-lib
* There are `UnmarshalX` methods which generally return the object, any remaining data and an error
* The remaining data is needed because the `AuthenticatorData` contains a variable-length CBOR-encoded map before the extension data, so we won't know where the extension data starts until after unmarshaling the `AttestedCredentialData`. A similar pattern can be seen in packages like the `pem` package: https://pkg.go.dev/encoding/pem#Decode
* Add linting

### CODE REVIEW

File: authenticator_flags.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+// ExtensionDataIncluded returns true if the AuthenticatorData has extension data.
+func (flags AuthenticatorFlags) ExtensionDataIncluded() bool {
+	return (flags & authenticatorFlagsED) == authenticatorFlagsED
+}
+
+const (
+	authenticatorFlagsUP = 1 << iota
+	_
```
Comment (desimone) [inline authenticator_flags.go:31]: TIL you coud do this!

File: attestation_object.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	// incorporated into the attestation statement by the authenticator.
+	Format string `json:"fmt"`
+	// The Statement is a signed data object, containing statements about a public key credential itself and
+	// the authenticator that created it.
+	Statement map[string]interface{} `json:"attStmt,omitempty"`
+}
+
+// UnmarshalAttestationObject unmarshals an attestation object from a slice of bytes. It expects a CBOR-encoded
+// map.
+func UnmarshalAttestationObject(raw []byte) (attestationObject *AttestationObject, remaining []byte, err error) {
```
Comment (desimone) [inline attestation_object.go:30]: We should probably add a `io.LimitReader` to guard against _very_ large / indefinite length paylods. I'm not sure of the best place to do it.

https://github.com/fxamacker/cbor#limitations

**Review by desimone — APPROVED** [review]

File: attestation_object.go (hunk 2 above)
Comment (calebdoxsey) [reply attestation_object.go:30]: It accepts a slice of bytes not a reader. What's the value of using a limit reader when it's already limited to the size of the slice of bytes?

Comment (desimone) [reply attestation_object.go:30]: Ah, fair point.

**Pull request merged**